k8sniff build automation

bcle opened the pull request:
The "push" target creates the binary, image, and pushes a versioned and tagged image to platform9/k8sniff on docker hub. It also creates a tag file that can be used as an artifact in a build pipeline.

TC build config: http://teamcity.platform9.horse:8111/admin/editBuild.html?id=buildType:Pf9project_DeccoBuilds_K8sniff

patform9 commented on Makefile:
```diff
@@ -1,46 +1,67 @@
 $(BIN_DIR):
 	mkdir -p $@
 
+$(VENDOR_DIR): $(DEP_EXE)
```
to be more accurate, I'd declare the Gopkg.lock/Gopkg.toml file as one of the prereqs for the vendor/ dir as well

patform9 commented on Makefile:
```diff
@@ -1,46 +1,67 @@
 	cd $(SRC_DIR)/cmd/k8sniff && go build -gcflags='-N -l' -o $${GOPATH}/bin/k8sniff-dbg
 
-$(K8SNIFF_EXE): | $(BIN_DIR)
+$(K8SNIFF_EXE): | $(BIN_DIR) $(GO_DEPS) $(VENDOR_DIR)
```
with the currently declared packages in `GO_DEPS` already part of Gopkg.toml, is declaring them as explicit prereqs necessary with `$(VENDOR_DIR)` already a prereq and the source files already under it? Wouldn't you then have unnecessary checkouts of them in the `GOSRC` folder? it seems like you can remove `GO_DEPS` altogether.

bcle commented:
@patform9 : removed the unnecessary go deps stuff, and added logic for logging into docker before pushing. Successful build with artifact: http://teamcity.platform9.horse:8111/viewLog.html?buildId=1037799&buildTypeId=Pf9project_DeccoBuilds_K8sniff&tab=artifacts
